src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -819,4 +819,39 @@ public function getEntityHistory($className, $id)
 
         return $result;
     }
+    
+    public function findRevisionsByDateAndUsername($className, $id, \DateTime $dateFrom = null, \DateTime $dateTo = null, $username = null)
```

Comment:
This is quite a large method signature, how about using a criteria object for the additional, optional, filters?

e.g.

```php
<?php

public function matchRevisions($className, $id, RevisionCriteria $criteria)
{
}

// ...

class RevisionCriteria
{
    public $dateFrom;
    public $dateTo;
    public $username;
}
```